oslo_vmware/api.py

```diff
@@ -259,8 +280,310 @@ def _create_session(self):
+            self._property_collector_version = ""
+            self._running_tasks = {}
+            self._pending_tasks = eventlet.Queue()
+
+            try:
+                pc = self.vim.service_content.propertyCollector
+                self._property_collector = self.vim.CreatePropertyCollector(pc)
+                LOG.debug("Created property collector: %s",
+                          vim_util.get_moref_value(self._property_collector))
+            except Exception:
+                LOG.exception("Failed to create property collector.")
+                self._property_collector_running = False
+                raise
+
+            self._property_collector_thread = eventlet.spawn(
+                self._property_collector_loop)
+
+    def _property_collector_loop(self):
+        LOG.debug("Property collector thread started.")
+
+        while self._property_collector_running:
+            with self._property_collector_lock.read_lock():
+                has_tasks = len(self._running_tasks) > 0
+
+            pending_task = None
+            if not has_tasks:
+                LOG.debug("No running tasks, waiting for new tasks...")
+                pending_task = self._pending_tasks.get()
+                # Sentinel for shutdown
+                if pending_task is None:
+                    break
+
+            with self._property_collector_lock.write_lock():
+                if pending_task is not None:
+                    task_value, future, ctx, task_filter = pending_task
+                    self._running_tasks[task_value] = (future, ctx,
+                                                       task_filter)
+                    LOG.debug("Received task %s, moved to running.",
+                              task_value)
+
+                while not self._pending_tasks.empty():
+                    task_data = self._pending_tasks.get_nowait()
+                    if task_data is None:  # Shutdown sentinel
+                        break
+
+                    task_value, future, ctx, task_filter = task_data
+                    self._running_tasks[task_value] = (future, ctx,
+                                                       task_filter)
+
+                if not self._property_collector_running:
+                    break
+
+                self._wait_for_task_updates()
+
+        LOG.debug("Property collector thread stopped.")
+
+    def _wait_for_task_updates(self):
+        """Waits 1 second for task updates using WaitForUpdatesEx."""
+        try:
+            options = self.vim.client.factory.create("ns0:WaitOptions")
+            options.maxWaitSeconds = 1
+
+            update_set = self.vim.WaitForUpdatesEx(
+                self._property_collector,
+                version=self._property_collector_version,
+                options=options)
+
+            if update_set:
+                self._property_collector_version = update_set.version
+                self._process_task_updates(update_set)
+        except Exception:
```

Comment:
Can you narrow the exception a bit down? This will also cover syntaxerrors and more.